### PR TITLE
Enable RedHat 6 in coreclr master

### PR DESCRIPTION
**Pull request by smile21prc**
Enable RedHat 6 in coreclr master.

This is identical as the approved PR to enable RedHat 6 in coreclr
release/2.0.0:
https://github.com/dotnet/coreclr/pull/13301

### CODE REVIEW

File: buildpipeline/pipelines.json [hunk 1]
```diff
@@ -24,6 +24,19 @@
           }
         },
         {
+          "Name": "DotNet-CoreClr-Trusted-Linux",
+          "Parameters": {
+            "DockerTag": "centos-6-783abde-20171304101322",
+            "Rid": "rhel6"
```
Comment (weshaggard) [inline buildpipeline/pipelines.json]: Shouldn't this be rhel.6?

File: buildpipeline/pipelines.json [hunk 2]
```diff
@@ -460,7 +473,42 @@
...
             "PB_BuildType": "Release"
           }
-        }
+        },
+        {
+          "Name": "Dotnet-CoreClr-Trusted-BuildTests",
+          "Parameters": {
+            "HelixJobType": "test/functional/cli/",
+            "TargetsWindows": "false",
+            "Rid": "rhel6-x64",
```
Comment (weshaggard) [inline buildpipeline/pipelines.json]: rhel.6-x64?

File: buildpipeline/pipelines.json (hunk 1 above)
Comment (weshaggard) [inline buildpipeline/pipelines.json]: also -x64?

File: buildpipeline/pipelines.json [hunk 3]
```diff
@@ -460,7 +473,42 @@
...
+            "Type": "build/product/",
+            "PB_BuildType": "Release"
+          }
+        },
+        {
+          "Name": "Dotnet-CoreClr-Trusted-BuildTests",
+          "Parameters": {
+            "HelixJobType": "test/functional/r2r/cli/",
+            "TargetsWindows": "false",
+            "Rid": "rhel6-x64",
```
Comment (weshaggard) [inline buildpipeline/pipelines.json]: rhel.6-x64

**Comment by jkotas** [comment]
@dotnet-bot test Windows_NT x64 Release Priority 1 Build and Test

File: buildpipeline/pipelines.json (hunk 1 above)
Comment (smile21prc) [inline buildpipeline/pipelines.json]: Thanks @weshaggard , changed it to rhel.6.

There is historical reason that in coreclr, the build and test parses RID in different ways. The RID you pass to the test build is the full RID like in dir.props, and in the product build it’s constructed from the OS portion & the arch portion.

This is a product build entry, so should be rhel.6 without -64.

File: buildpipeline/pipelines.json (hunk 2 above)
Comment (smile21prc) [inline buildpipeline/pipelines.json]: Changed as suggested.

File: buildpipeline/pipelines.json (hunk 3 above)
Comment (smile21prc) [inline buildpipeline/pipelines.json]: Changed as suggested.

File: build.sh [hunk 4]
```diff
@@ -398,7 +398,7 @@ isMSBuildOnNETCoreSupported()
         if [ "$__HostOS" == "Linux" ]; then
             __isMSBuildOnNETCoreSupported=1
             # note: the RIDs below can use globbing patterns
-            UNSUPPORTED_RIDS=("debian.9-x64" "ubuntu.17.04-x64" "alpine.3.6.*-x64" "rhel.6-x64" "")
+            UNSUPPORTED_RIDS=("debian.9-x64" "ubuntu.17.04-x64" "alpine.3.6.*-x64" "")
```
Comment (weshaggard) [inline build.sh]: This looks like a merge conflict you resolved. While I think it is the correct change why did you leave the empty quotes? I would remove those.

Comment (smile21prc) [inline build.sh]: Yep, it's a merge conflict I resolved. Good suggestion @weshaggard and I'll remove the empty quotes.

**Pull request merged**